### DISCUSSION TITLE
I cannot get my hyper key to work with the keymaps

### DISCUSSION THREAD

**Discussion by joelzanden** [zed-industries/zed#6660] [Feature Requests]
I have caps lock set up as a hyper key, with the help of Karabiner Elements, on my Mac. I want to do a keymap that triggers formatting of the current file (among many others). I have the following in my keymap.json file:
```json
[
  {
    "context": "Editor",
    "bindings": {
      "cmd-alt-shift-ctrl-f": "editor::Format"
    }
  }
]
```

But I cannot get this to work. Can anyone kindly point me in a constructive direction. This is literally the only thing keeping me from switching over from VS Code right now.

**Comment by tml** [comment]
Following the process at https://holmberg.io/hyper-key/ to setup my caps lock as a "hyper key", this behaviour works as expected for me in Zed 0.119.20

**Reply by joelzanden** [reply]
Thank you for your answer!
However, setting up the hyper key functionality on my system is not the issue I have. I have been a hyper key user for many years and I'm using it throughout my system.

This is how I'm currently remapping one hyper key combo in VS Code:
```json
{
    "key": "ctrl+shift+alt+cmd+f",
    "command": "editor.action.formatDocument",
    "when": "editorHasDocumentFormattingProvider && editorTextFocus && !editorReadonly && !inCompositeEditor"
}
```

Thing is, I cannot get the remap to work in Zed. It's probably a syntax thing. Any help would be greatly appreciated as I would like to switch over to Zed as my daily driver.

**Reply by tml** [reply]
I wasn't trying to offer instructions, I was saying "if yours is setup the same way this article says to set it up, I cannot reproduce your behaviour here; if yours is set up some other way, please give me steps to reproduce that setup here so I can try to reproduce it locally"

**Reply by joelzanden** [reply]
Thank you! Sorry for misreading your answer. The setup outlined in the blog post that you linked to is exactly what I have on my system. How do your mappings in the Zed key map settings look?

**Reply by tml** [reply]
```
[
  {
    "context": "Editor",
    "bindings": {
      "ctrl-shift-alt-cmd-f": "editor::Format"
    }
  }
]
```

**Reply by tml** [reply]
I installed Keycastr to grab a screenshot showing that Karabiner is sending the "hyper key" as expected and then recorded this working on my local machine: 

https://github.com/zed-industries/zed/assets/35148/0391f443-f3f2-4e2f-b0e2-f72c7a66e834

**Reply by joelzanden** [reply]
Wow, thank you! Now it works just as expected.

The only thing differing between our two json snippets seems to be the order in wich the keys are written. And when I now revert back to my original snippet, that also works. I should have taken note of the Zed version I was using when I created this issue. Or maybe it has something to do with me trying this out on both an Intel based mac as well as a M1 mac.

Anyways, it works now. I'll set up my key maps and switch to Zed.
